### PR TITLE
Update codeflare component description

### PR DESCRIPTION
**Pull request by ronensc**
## Description

After a discussion with @anishasthana, it was discovered that the description of the CodeFlare component in the manifests is not up-to-date. The current description indicates that the CodeFlare Operator should be installed in advance, but in reality, it will be installed on the fly.
This PR updates the description

## Merge criteria:

- [ ] The commits are squashed in a cohesive manner and have meaningful messages.
- [ ] Testing instructions have been added in the PR body (for PRs involving changes that are not immediately obvious).
- [ ] The developer has manually tested the changes and verified that the changes work

### CODE REVIEW

File: apis/datasciencecluster/v1/datasciencecluster_types.go [hunk 1]
```diff
@@ -46,7 +46,7 @@ type Components struct {
 	Workbenches workbenches.Workbenches `json:"workbenches,omitempty"`
 
 	// ModelMeshServing component configuration.
-	// Require CodeFlare Operator to be installed before enable component
+	// If CodeFlare Operator is not already installed, it'll be installed on the fly.
```
Comment (zdtsw) [inline apis/datasciencecluster/v1/datasciencecluster_types.go:49]: i think the old comments is wrong, should be removed, modelmesh does not need codeflare as pre-req

File: apis/datasciencecluster/v1/datasciencecluster_types.go [hunk 2]
```diff
@@ -60,11 +60,11 @@ type Components struct {
 	Kserve kserve.Kserve `json:"kserve,omitempty"`
 
 	// CodeFlare component configuration.
-	// Require CodeFlare Operator to be installed before enable component
+	// If CodeFlare Operator is not already installed, it'll be installed on the fly.
 	CodeFlare codeflare.CodeFlare `json:"codeflare,omitempty"`
 
 	// Ray component configuration.
-	// Require CodeFlare Operator to be installed before enable component
+	// If CodeFlare Operator is not already installed, it'll be installed on the fly.
```
Comment (zdtsw) [inline apis/datasciencecluster/v1/datasciencecluster_types.go:67]: same here, should not be needed

File: apis/datasciencecluster/v1/datasciencecluster_types.go [hunk 3]
```diff
@@ -60,11 +60,11 @@ type Components struct {
 	Kserve kserve.Kserve `json:"kserve,omitempty"`
 
 	// CodeFlare component configuration.
-	// Require CodeFlare Operator to be installed before enable component
+	// If CodeFlare Operator is not already installed, it'll be installed on the fly.
```
Comment (zdtsw) [inline apis/datasciencecluster/v1/datasciencecluster_types.go:63]: I think it should be:
If CodeFlare Operator has been installed in the cluster, it should be uninstalled first before enabled component.

File: bundle/manifests/opendatahub-operator.clusterserviceversion.yaml [hunk 4]
```diff
@@ -73,9 +73,9 @@ metadata:
           "metadata": {
             "labels": {
               "app.kubernetes.io/created-by": "opendatahub-operator",
-              "app.kubernetes.io/instance": "default",
+              "app.kubernetes.io/instance": "default-feature",
```
Comment (zdtsw) [inline bundle/manifests/opendatahub-operator.clusterserviceversion.yaml:76]: this probably was my fault, i pointed out to update them in the previous PR but not done it properly in the source config/samples/featurectracker_v1_features.yaml, thats why you got it reverted after generate manifests.
could you update that one instead ?

File: apis/datasciencecluster/v1/datasciencecluster_types.go (hunk 1 above)
Comment (ronensc) [reply apis/datasciencecluster/v1/datasciencecluster_types.go]: done

File: apis/datasciencecluster/v1/datasciencecluster_types.go (hunk 2 above)
Comment (ronensc) [reply apis/datasciencecluster/v1/datasciencecluster_types.go]: done

File: apis/datasciencecluster/v1/datasciencecluster_types.go (hunk 3 above)
Comment (ronensc) [reply apis/datasciencecluster/v1/datasciencecluster_types.go]: done

File: bundle/manifests/opendatahub-operator.clusterserviceversion.yaml (hunk 4 above)
Comment (ronensc) [reply bundle/manifests/opendatahub-operator.clusterserviceversion.yaml]: I just run `make bundle` and this change popped up. I modified `config/samples/featurectracker_v1_features.yaml` to match the bundle's values. Please LMK if this is what you meant.

**Review by zdtsw — APPROVED** [review]
/lgtm

**Pull request merged**